src/main/java/io/cdap/delta/datastream/DatastreamEventConsumer.java

```diff
@@ -252,6 +258,26 @@ private void fetchNext() throws Exception {
     }
   }
 
+  private List<SortKey> getSortKeys(GenericRecord record) {
+    GenericArray<?> keys = (GenericArray<?>) record.get(SORT_KEYS_FIELD_NAME);
+    List<SortKey> sortKeys = null;
+    if (keys != null) {
+      sortKeys = new ArrayList<>(keys.size());
+      for (Object key : keys) {
+        if (Long.class.equals(key.getClass())) {
+          sortKeys.add(new SortKey(Schema.Type.LONG, key));
+        } else if (Integer.class.equals(key.getClass())) {
+          sortKeys.add(new SortKey(Schema.Type.INT, key));
+        } else if (key instanceof CharSequence) {
+          sortKeys.add(new SortKey(Schema.Type.STRING, key.toString()));
+        } else {
+          throw new IllegalArgumentException("Unsupported sort key type: " + schema.getType());
```

Comment:
`schema` here is not the type of the sort key

Reply:
Fixed